content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md

```diff
@@ -1249,31 +1249,32 @@ scheduling pods:
   get rejected by the existing webhook server since the `env` label is unset, and the replacement Pod
   cannot be created. Eventually, the entire Deployment for the webhook server may become unhealthy.
 
-If you use admission webhooks to check Pods, consider excluding the namespace where your webhook
-listener is running, by specifying a
+  If you use admission webhooks to check Pods, consider excluding the namespace where your webhook
+  listener is running, by specifying a
   [namespaceSelector](#matching-requests-namespaceselector).
 
 * If the cluster has multiple webhooks configured (possibly from independent applications deployed on
   the cluster), they can form a cycle.  Webhook A must be called to process startup of webhook B's
-  pods and vice versa.  If both webhook A and webhook B ever become inactive at the same time (for
+  pods and vice versa. If both webhook A and webhook B ever become unavailable at the same time (for
   example, due to a cluster-wide outage or a node failure where both pods run on the same node)
   deadlock occurs because neither webhook pod can be recreated without the other already running.
 
-  It is recommended to establish a desired order for webhooks to start, then to exclude "earlier"
-  webhooks' resources from being inspected by later webhooks.  This ensures that the "earliest"
-  webhook can start, which in turn allows the next.
+  One way to prevent this is to establish a desired order for webhooks to start, then to exclude 
+  "earlier" webhooks' resources from being inspected by "later" webhooks.  This ensures that the 
+  "earliest" webhook can start, which in turn allows the next.
```

Comment:
I don't understand how this could work and don't think it's an effective control. See https://stackoverflow.com/a/69166466 for informal confirmation that admission webhooks are _not_ ordered.

Reply:
I think you've taken the wrong idea from that paragraph, which suggests I could explain it more clearly!  I'm talking about the order in which the webhook pods start rather than the order that webhooks apply to a particular resource.

The deadlock that can happen is if 

* webhook A is unavailable
* webhook B is unavailable
* webhook A is needed to approve webhook B's pod
* webhook B is needed to approve webhook A's pod.

So the suggestion is:

* Pick the webhook that needs to apply most broadly.  Let's say that's A
* Make sure that no other webhooks can interfere with starting A.
* Therefore A can always start, even if no other webhooks are available
* Therefore B can start, because it depends on A but we know that A can start.

If you have3 webhooks, A, B, C, then

* Make sure B and C exclude A's pods
* Therefore A can start
* Make sure C excludes B's pods
* Therefore B can start, since we know that A can start
* Therefore C can start, since A and B can start.

Reply:
I'll try to reword it, but suggestions also welcome

Reply:
This kind of workload sequencing doesn't really hold because nodes are not considered durable and resilient. Also, do try to reword.

Reply:
I get the point we want to make. Maybe that's worth shifting to its own point (eg a blog article)?

Reply:
I think as long as you've got an ordering, node failure is fine:
* webhook A comes up
* webhook A's node dies
* webhook A is rescheduled somewhere else
* now webhook B can start... and so on.

Reply:
It should get there _eventually_, whereas without some sort of ordering like this, node failures will cause deadlocks and you'll be forced to intervene by temporarily removing your webhook